File: Dataset/Leetcode/train/56/559.java
 class Solution {
    public int[][] XXX(int[][] intervals) {
       //定义排序规则
        Arrays.sort(intervals, (o1, o2) -> o1[0] == o2[0] ? o1[1] - o2[1] : o1[0] - o2[0]);
        LinkedList<int[]> res = new LinkedList<>();
        int[] pre = intervals[0];
        for (int[] i : intervals) {
            if (i[0] > pre[1]) {
                res.add(pre);
                pre = i;
            } else {
                pre[1] = i[1] > pre[1] ? i[1] : pre[1];
            }
        }
        res.add(pre);
        return res.toArray(new int[res.size()][]);
    }
}

File: Dataset/Leetcode/train/7/559.py
 class Solution:
   def XXX(self, x):
       """
       :type x: int
       :rtype: int
       """
       if x > (2**31-1) or x < -2**31:
           return 0
       a = str(x)
       if a[0] == '-':
           x = a[1:]
           x = x[::-1]
           x = -int(x)
       else:
           x = int(a[::-1])
       if x > (2**31-1) or x < -2**31:
           return 0
       return x

